martin-core/src/resources/sprites/mod.rs

```diff
@@ -71,27 +71,273 @@ impl SpriteSources {
 
     /// Adds a sprite source directory containing SVG files.
     /// Files are ignored - only directories accepted. Duplicates ignored with warning.
+    /// Performs basic validation of SVG format before adding the source.
     pub fn add_source(&mut self, id: String, path: PathBuf) {
         let disp_path = path.display();
+
         if path.is_file() {
             warn!("Ignoring non-directory sprite source {id} from {disp_path}");
-        } else {
-            match self.0.entry(id) {
-                Entry::Occupied(v) => {
+            return;
+        }
+        if !path.exists() {
+            warn!("Sprite source {id} path doesn't exist: {disp_path}");
+            return;
+        }
+
+        match std::fs::read_dir(&path) {
```

Comment:
Let's move this validation to a new function. It is kind of messy to read otherwise